mobile/App.js

```diff
@@ -6,25 +6,111 @@
  * @flow strict-local
  */
 
-import React, {useState} from 'react';
+import React, {useState, useRef, useEffect} from 'react';
 import {SafeAreaView} from 'react-native-safe-area-context';
+import AsyncStorage from '@react-native-async-storage/async-storage';
+
 import HomeView from './src/views/HomeView';
-import ScanView from './src/views/ScanView';
 import VendorView from './src/views/VendorView';
 import RegistrationView from './src/views/RegistrationView';
 import {NavigationContainer} from '@react-navigation/native';
 import {createBottomTabNavigator} from '@react-navigation/bottom-tabs';
 import ScreenOption from './src/utilites/ScreenOption';
-import {globalStyleSheet} from './src/utilites/Theme';
 import MyICardView from './src/views/MyICardView';
-
+import { AppState } from "react-native";
 import AuthContext from './src/context/AuthContext';
+import { storeDate } from './src/utilites/StoreDate';
+import { storeUser } from './src/utilites/StoreUser';
 
 const Tab = createBottomTabNavigator();
 
 const App = () => {
   const [user, setUser] = useState(null);
+  const [check, setCheck] = useState(false);
   const value = {user, setUser};
+  const appState = useRef(AppState.currentState);
+  
+  const logoutCheck = async(userData = null,startup = false) => {
+    
+    try{
+        const lastDate = parseInt(await AsyncStorage.getItem('lastOpened'));    
+        if(lastDate != null){
+            //change num in if statement below to a month in milliseconds
+            if(new Date().getTime() - lastDate > 2629800000){
+                setUser(null)
+                storeUser(null)
+            }
+            else{
+                if(startup){
+                  setUser(userData)
+                }
+                await storeDate()
+            }
+        }
+    } catch(error){
+        logoutCheck()
+    }
+}
+  useEffect(() => {
+    const checkUser = async() => {
+      const userData = JSON.parse(await AsyncStorage.getItem('userData'))
+      if(userData != null){
+        logoutCheck(userData, true)
```

Comment:
We can remove the if statement and always pass user, then with the removal of the startup flag, it won't trigger the if statement.